### PR TITLE
Check for version of 'None'

### PR DESCRIPTION
**Pull request by dvf**
- [ ] Added **tests** for changed code. **I wasn't sure how to write tests for this, any help would be most appreciated!**
- [ ] Updated **documentation** for changed code. 
---

I haven't added tests because I wasn't sure how to test for something like this, any help would be appreciated 🙏. Linked to #817

### CODE REVIEW

File: poetry/semver/version.py [hunk 1]
```diff
@@ -197,6 +197,11 @@ def include_max(self):
     @classmethod
     def parse(cls, text):  # type: (str) -> Version
         match = COMPLETE_VERSION.match(text)
```
Comment (DanCardin) [inline poetry/semver/version.py]: You'll need to delete this line first, it seems

Comment (dvf) [reply poetry/semver/version.py]: Wait why? It's only failing in Python 2.7

Comment (DanCardin) [reply poetry/semver/version.py]: What I mean to say is that, line 199 `match = COMPLETE_VERSION.match(text)` is the same as line 201 inside the `try`/`catch`.

So when `.match` raises a `TypeError` it never makes it to the `try`/`catch`ed version.

I deleted line 199, and it appeared to fix https://github.com/sdispater/poetry/issues/837

**Comment by DanCardin** [comment]
this one was superceded by https://github.com/sdispater/poetry/pull/917, so can be closed

**Pull request closed without merging**